src/transformers/models/canine/modeling_canine.py

```diff
@@ -1130,12 +1139,12 @@ def forward(
 
         # We can provide a self-attention mask of dimensions [batch_size, from_seq_length, to_seq_length]
         # ourselves in which case we just need to make it broadcastable to all heads.
-        extended_attention_mask: torch.Tensor = self.get_extended_attention_mask(attention_mask, input_shape)
+        extended_attention_mask: torch.Tensor = self.get_extended_attention_mask(attention_mask, input_shape, device)
```

Comment:
It looks like you inserted an extra argument here?

Reply:
Hi Matt, sorry actually i have not added it may be it got added by copilot or may be during "make fixup"

Reply:
ok so i've checked it and i got to know that this function is having this argument `device` which you can see in the signature 
![image](https://user-images.githubusercontent.com/36916536/166694480-e533981e-275c-4ac8-802f-e0ca36c23e2a.png)
 & since this `device` was assigned in this so i guess the copilot took it. 
![image](https://user-images.githubusercontent.com/36916536/166694813-ef9fcbb6-4978-40f8-b8db-1acce34f4dc9.png)
 if you see in the first line, the varible is being used/assigned . 
but i will go with your suggestion if you say to remove it , so shall i remove?

Reply:
I think it's safer to remove it, since the code worked beforehand! Very interesting that copilot is picking up on these things, though - did you use it to write these type hints?

Reply:
No, for type hints, I was not relying on it as it's something that is confusing for this tool. so, I better thought to write these types on my own by cross-checking with the docstring written in the class & the models/functions being referred there ! though it was showing for some but it got messed up with hints totally !!! its still not smarter than human though 😄